pkg/imgpkg/imageset/image_set.go

```diff
@@ -146,27 +148,40 @@ func checkForAnyAsyncErrors(imgOrIndexes []imagedesc.ImageOrIndex, errCh chan er
 	return nil
 }
 
-func (i ImageSet) getImageOrImageIndexForMultiWrite(item imagedesc.ImageOrIndex, importRepo regname.Repository, registry registry.ImagesReaderWriter) (regname.Tag, regremote.Taggable, error) {
-	uploadTagRef, err := util.BuildDefaultUploadTagRef(item, importRepo)
+func (i ImageSet) getImageOrImageIndexForMultiWrite(item imagedesc.ImageOrIndex, importRepo regname.Repository, registry registry.ImagesReaderWriter, useRepoBasedTags bool) ([]regname.Tag, []regremote.Taggable, error) {
+	uploadTagRef1, err := util.BuildDefaultUploadTagRef(item, importRepo)
 	if err != nil {
-		return regname.Tag{}, nil, err
+		return []regname.Tag{}, nil, err
 	}
 
-	var artifactToWrite regremote.Taggable
-	switch {
-	case item.Image != nil:
-		artifactToWrite, err = i.mountableImage(*item.Image, uploadTagRef, registry)
+	tagArr := make([]regname.Tag, 1, 2)
```

Comment:
I think we talked about only having 1 tag, so we do not need a slice here

Reply:
Got it. That's because of my comment: https://github.com/vmware-tanzu/carvel-imgpkg/issues/331#issuecomment-1046136094